crates/shared/src/sources/balancer_v2/event_handler.rs

```diff
@@ -156,93 +153,44 @@ impl BalancerPoolRegistry {
             .lock()
             .await
             .store
-            .stable_pools_for(pool_ids)
+            .pools_by_id(pool_ids)
     }
 }
 
 #[async_trait::async_trait]
-impl EventStoring<WeightedPoolFactoryEvent> for PoolStorage {
+impl<Factory> EventStoring<BasePoolFactoryEvent> for PoolStorage<Factory>
+where
+    Factory: FactoryIndexing,
+{
     async fn replace_events(
         &mut self,
-        events: Vec<EthContractEvent<WeightedPoolFactoryEvent>>,
+        events: Vec<EthContractEvent<BasePoolFactoryEvent>>,
         range: RangeInclusive<BlockNumber>,
     ) -> Result<()> {
-        self.replace_events_inner(
```

Comment:
Can this method be removed now?

I think one reason why we differentiated between replace and delete + insert was that the db transaction could be atomic so we don't end up in a state where past blocks are detailed but new blocks are not applied. Not sure how likely/relevant this is though compared to the added complexity.

Reply:
Right - we now just implement the logic in 
```rust
fn replace_events() {
    // logic here...
}
```

Instead of 
```rust
fn replace_events_inner() {
    // logic here...
}
fn replace_events() {
    replace_events_inner();
}
```

Since I found it was easier to implement `PoolStorage` in terms of:
- Add events
- Remove events newer than block

And test those two methods interdependently instead of testing `replace_events_inner` assuming that it called the two above methods.